### PR TITLE
docs(sdk): update contributing guidelines

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
- Updates and deduplicates README information to help contributors.
- Adds dockerized code formatting/checking scripts to kfp README.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/README.md [hunk 1]
```diff
@@ -1,4 +1,45 @@
...
+
+#### Formatting [Required]
+Please format your code using [yapf](https://github.com/google/yapf) according to the [`.style.yapf`](https://github.com/kubeflow/pipelines/blob/master/.style.yapf) file.
+
+From the project root, run the following code to format your code:
+```sh
+docker container run \
+  -v $PWD:/root/pipelines \
+  python:3.7-slim \
+  /bin/bash -c "pip install yapf && yapf --in-place --recursive /root/pipelines/sdk/python"
```
Comment (chensun) [inline sdk/README.md:21]: Should we just give the `yapf` command? Every KFP SDK developer must have Python environment already, installing `yapf` via pip isn't much extra work (we could even include it in some `requirements-dev.txt` for example). On the other hand, they may not necessarily have docker runtime installed or running all the time.

Comment (chensun) [reply sdk/README.md:21]: Same argument for `pylint` and `docformatter`

**Review by chensun — COMMENTED** [review]
Thanks Connor for putting this up!

File: sdk/README.md (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/README.md]: Thanks for the suggestion, I agree.

Jointly, [70076a2](https://github.com/kubeflow/pipelines/pull/7436/commits/70076a2d50dcc23a67ee365ddab551f6e056a625), [d5b1364](https://github.com/kubeflow/pipelines/pull/7436/commits/d5b136479320d6b97e23b442d899df5c43e63d57), and [ddc6d41](https://github.com/kubeflow/pipelines/pull/7436/commits/ddc6d413718cf04c3852dd3a19f6b7ec0cce727e) achieve this change.

**Comment by ji-yaqi** [comment]
Thanks Connor! 
/lgtm

File: sdk/README.md [hunk 2]
```diff
@@ -1,4 +1,48 @@
...
+```sh
+yapf --in-place --recursive ./sdk/python
+```
+
+#### Pylint [Encouraged]
+We encourage you to lint your code using [pylint](https://pylint.org/) according to the project [`.pylintrc`](https://github.com/kubeflow/pipelines/blob/master/.pylintrc) file.
+
+From the project root, run the following code to lint your code:
+```sh
+pylint ./sdk/python/kfp ./sdk/python/tests/
```
Comment (chensun) [inline sdk/README.md:28]: nit: We can skip `./sdk/python/tests/` here, these are tests only for the now deprecated folder. We don't expect contributors to touch the code or its test again.

**Review by chensun — COMMENTED** [review]
/lgtm

Thanks!

File: sdk/README.md (hunk 2 above)
Comment (connor-mccarthy) [reply sdk/README.md]: Updated

**Comment by chensun** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**